test/object-store/nested_collections.cpp

```diff
@@ -0,0 +1,83 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include <catch2/catch_all.hpp>
+
+#include "util/test_file.hpp"
+#include "util/test_utils.hpp"
+#include "util/event_loop.hpp"
+#include "util/index_helpers.hpp"
+
+#include <realm/object-store/binding_context.hpp>
+#include <realm/object-store/list.hpp>
+#include <realm/object-store/object.hpp>
+#include <realm/object-store/object_schema.hpp>
+#include <realm/object-store/property.hpp>
+#include <realm/object-store/results.hpp>
+#include <realm/object-store/schema.hpp>
+
+#include <realm/object-store/impl/realm_coordinator.hpp>
+#include <realm/object-store/impl/object_accessor_impl.hpp>
+
+#include <realm/version.hpp>
+#include <realm/db.hpp>
+
+#include <cstdint>
+
+using namespace realm;
+
+TEST_CASE("nested-list") {
+    InMemoryTestFile config;
+    config.cache = false;
+    config.automatic_change_notifications = false;
+    auto r = Realm::get_shared_realm(config);
+    r->update_schema(
+        {{"list_of_list", {{"nested_list", {PropertyType::Array, PropertyType::Array, PropertyType::Int}}}}});
```

Comment:
I am not sure what is the best way to express this at schema level. 
Doing something like `Array | Array | Int` won't work, since it will be the same of doing `128|128|0` which is equal to 128 (thus there will be no way to identify eventual nested collections).

Another possibility could be to add an `Append or Add` method to `ProperyType` and implement the proper bitwise mathematic in order to nest the collection. But I would avoid this, since `Array|Array|Int` would equal to 256 which is the same numeric value of `Sets`.

Reply:
going with an explicit of nested collections.